FirebaseSessions/Sources/FirebaseSessionsError.swift

```diff
@@ -20,4 +20,6 @@ enum FirebaseSessionsError: Error {
   case SessionSamplingError
   /// Firebase Installation ID related error
   case SessionInstallationsError
+  /// Settings related error
+  case SettingsError(String)
```

Comment:
It probably isn't applicable yet, but I believe normally Swift recommends having a separate case for each error, so it's a bit more granular than Java where you have a class of errors with an error message that differentiates them. The reason Swift recommends granular enum cases is so you can add a switch on the error later on. We don't have a use case for that yet, but we could see a situation where we want to retry a 400-level error, but not a parsing error, for example.

I don't think it's necessary to make that change now but thought I'd explain.

Reply:
That's helpful to know. I guess I added the separation between each error in the `String` description passed in when creating the error. Is it preferable then to create individual errors for each case and not add any additional description?

Reply:
I think you'd do both. That way you get both a nice description and a way to do a switch on it.


Not something necessary for this PR